src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/PlatformProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Editor;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Configuration
+{
+    /// <summary>
+    /// Provides 'Platform' project configuration dimension and values.
+    /// </summary>
+    [Export(typeof(IProjectConfigurationDimensionsProvider2))]
+    [AppliesTo(ProjectCapabilities.ProjectConfigurationsDeclaredDimensions)]
+    [Order(int.MaxValue -1)]
+    internal class PlatformProjectConfigurationDimensionProvider : BaseProjectConfigurationDimensionProvider
+    {
+        [ImportingConstructor]
+        public PlatformProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlAccessor)
+            : base(projectXmlAccessor, ConfigurationGeneral.PlatformProperty, "Platforms")
+        {
+        }
+
+        /// <summary>
+        /// Modifies the project when there's a configuration change.
+        /// </summary>
+        /// <param name="args">Information about the configuration dimension value change.</param>
+        /// <returns>A task for the async operation.</returns>
+        public override async Task OnDimensionValueChangedAsync(ProjectConfigurationDimensionValueChangedEventArgs args)
```

Comment:
Correct, that's one of the downsides of adding an extensibility point with full control. To make things even more interesting, add/remove need to be saved at different times than rename due to the order of operations on the CPS side.